### PR TITLE
Workload wizard

### PR DESCRIPTION
**Pull request by mantis-toboggan-md**
#825 
One issue: the Tabbed component hash-change is interfering with workload navigation, cancelling the step change to steps that include tabs on first click. I included [this](https://github.com/rancher/dashboard/compare/master...mantis-toboggan-md:workload-wizard?expand=1#diff-2c697cc782dcebff7aebba33c6d4f8cfR149) workaround I'm not very happy with

Wes is working on a 'view yaml' option in ButtonDropdown so I'll add that when his code is merged.

Since there isn't a new detail design for workload and the wizard certainly doesn't make sense there, I preserved the old workload detail page, including the old create page used in it, for the time being.

Screen Shot 2020-07-26 at 8 59 26 PM

Screen Shot 2020-07-26 at 8 59 59 PM

Screen Shot 2020-07-26 at 9 01 36 PM

Screen Shot 2020-07-26 at 9 00 41 PM

### CODE REVIEW

**Comment by vincent99** [comment]
Step 1:
  - I thought the boxes were going to be somehow more graphical and describe the differences between the types so you know which one you should pick?  
  - For Apps, Will wanted it not to be a "wizard" until after step 1, so it's just the boxes and clicking on them is the equivalent of next.  Should this work the same way? (@deniseschannon)
image
  - Drop RC and RS, they're not commonly created on their own (Deployments create ReplicaSets, and ReplicationControllers are an early form of Deployment)
  - Boxes should get border and cursor on hover so you know you can click them (like Apps does, see above)

Step 2:
  - Show/Hide host port link: `class="hand"`
  - Deployment replicas or CronJob schedule would make more sense to me in the name section, they're not really related to the image.
  - Turn off adding a label & annotation row by default for consistency with Port & Env
  - Environment vars might as well be its own section now and should have header the same size as the others
  - Some validation is present, but invisible?  If you just click next on step 2 nothing happens and nothing tells you why.  But if you fill in name/image then next changes to a dropdown and works (the dropdown is funky, but I think @westlywright is working on that).

Step 3:
  -  ``
  - If this isn't going to make it in this step should just tell you that it's coming soon and to use the review yaml option to add what you need.

Step 4:
  - Should be able to click on the entire healthcheck/networking/etc "tab" and the padding around the text to switch between them (not just the small `<a>` area)
  - Highlight of the selected "tab" should probably be more prominent (@lvuch)
  - "Default" is not the default `dnsPolicy`, [despite the name](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-dns-policy).    It should default to `ClusterFirst`.
  - Upgrade policy radios are in a span-6 and don't need to be.
  - Pod scheduling fonts are bolder than others
  - Resources tab is pretty packed, Node and Pod scheduling could probably each be their own section?

Misc:

- Cancel goes back to step 1 instead of out of the wizard

- Not sure I really like the blue section between the steps and the content that restates 3 things that are an inch away.  Maybe wizards shouldn't get the "Deployment: Create" at the top or something.
image

- Not directly related or limited to this PR but button styles and labels are inconsistent.
   - I think all the add port/env/label/annotation buttons should be:
      - the transparent blue
      - the same size
      - consistently margined from the content above (add port has extra space)
      - and say `Add <singular of the thing the title says>`
    - The `Next` in the footer should be solid blue (on every step), and the only solid blue button, because it's the primary action.
    - Also I don't know why REMOVE buttons are yelling, or not styled as buttons. @lvuch

**Comment by deniseschannon** [comment]
Step 1:

Here are the desciptions we should put under each type and include the links. 
```
Deployment:
A [deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/) is a set of multiple, identical pods that run on any node. Lifecycle management of the pods are defined explicitly in the deployment including but not limited to how many pods should be deployed and which nodes the pods should be deployed to.

DaemonSet: 
A [DaemonSet](https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/) ensures that a pod runs on all Kubernetes nodes. As new nodes are added to the cluster, a pod automatically gets added onto any new node. As nodes are removed from the cluster, the pods are garbage collected. 

StatefulSet:
A [StatefulSet](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/) manages the deployment and scaling of a set of pods. It guarantees ordering and uniqueness of pods. If you want to use storage volumes for persistence for your workload, it’s recommended to use a StatefulSet. 

Job: 
A [Job](https://kubernetes.io/docs/concepts/workloads/controllers/job/) creates one or more pods and ensures that a specified number of them successfully terminate. After a specified number of successful completions is reached, the Job is complete and no additional pods are created. 

 CronJob:
A [CronJob](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/) creates jobs on a repeating schedule. It runs the jobs on a periodic schedule written in [Cron](https://en.wikipedia.org/wiki/Cron) format. The schedule of a CronJob is based on the timezone of the `kube-controller-manager`. 
 ```

I am leaning towards the wizards for resources *should* show Step 1 and the wizard part. I think the App use case is a bit more nuanced and unique. If there are complaints, we can change it later. 

Misc Comments:

For the wizard header bar that seems redundant, the goal was originally to put more descriptive text, but I hadn't defined that text for workload. Let's try and add this in and see if it makes it less repetitive. 

```
 Step 1: 
Select the type of [workload](https://kubernetes.io/docs/concepts/workloads/). There are several types of workloads that will deploy pods on to nodes. Each type has unique properties to help define and maintain the lifecycle of the pods.

Step 2: 
Define the container of the pod. Each workload type ultimately deploys pods, which are the smallest unit of computing that can be created and managed in Kubernetes. The main definition of the pod is the container. These are the basic fields defined on a container, but there are more fields that can be defined in the Advanced step or in the yaml.

Step 3:
Define [storage](https://kubernetes.io/docs/concepts/storage/) for the container. Persist and share date separate from the lifecycle of the individual container. 

Step 4: 
Define the advanced options. Each pod has additional options that can be defined as part of the container, part of the pod and part of the workload type. Rancher recommends reading more about these options before attempting to configure these. 
```

@vincent99 Can you help review the text and edit for wordsmithing/accuracy? 


@mantis-toboggan-md Let's make a separate issue regarding the labels/buttons styling.

**Comment by vincent99** [comment]
Note: target=_blank, rel=nofollow, noopener, norefererr on all links...

Suggested edits (instead of editing your copy since pretty substantial):

- [Deployments](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/) run a scalable number of replicas of a pod distributed among the eligible nodes.  Changes are rolled out incrementally and can be rolled back to the previous revision when needed.  Recommended for stateless & horizontally-scalable workloads.

- [StatefulSets](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/) manage stateful applications and provide guarantees about the ordering and uniqueness of the pods created. Recommended for workloads with persistent storage or strict identity, quorum, or upgrade order requirements.

- [DaemonSets](https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/) run exactly one pod on every eligible node.  When new nodes are added to the cluster, DaemonSets automatically deploy to them.  Recommended for system-wide or vertically-scalable workloads that never need more than one pod per node.

- [Jobs](https://kubernetes.io/docs/concepts/workloads/controllers/job/) create one or more pods to reliably perform a one-time task by running a pod until it exits successfully.  Failed pods are automatically replaced until the specified number of completed runs has been reached.  Jobs can also run multiple pods in parallel or function as a batch work queue.

- [CronJobs](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/) create Jobs, which then run Pods, on a repeating schedule. The schedule is expressed in standard Unix [cron](https://en.wikipedia.org/wiki/Cron) format, and uses the timezone of the Kubernetes control plane (typically UTC).

--

Step 1: Select the type of [workload](https://kubernetes.io/docs/concepts/workloads/) to create. Each type has unique properties that define where the pods will be run, how they will be maintained or scaled, and what happens when they fail.

Step 2: 
Define the container of the pod. Each workload type ultimately deploys pods, which are the smallest unit of compute managed by Kubernetes. The main definition of the pod is the container. These are the basic fields defined on a container, but there are more fields that can be defined in the Advanced step or in the raw yaml.

Step 3:
Pods only have their own local ephemeral storage by default, which will be wiped out when the pod dies.  To persist data that survives beyond the life of a single pod, you can attach external [storage](https://kubernetes.io/docs/concepts/storage/) (e.g. block devices or NFS shares) from a provider and make it available to the pod.

Step 4: 
Define the advanced options. Each pod has additional options that can be defined as part of the container, part of the pod and part of the workload type. We recommends reading the detailed Kubernetes documentation before attempting to configure these.

**Comment by vincent99** [comment]
It is not at all clear to me how this amount of text for the workload or service types fits into the button style shown..

**Comment by lvuch** [comment]
Screen Shot 2020-07-28 at 9 08 18 AM

**Comment by mantis-toboggan-md** [comment]
Screen Shot 2020-07-28 at 4 03 00 PM
Screen Shot 2020-07-28 at 4 03 15 PM
Screen Shot 2020-07-28 at 4 05 34 PM

I disabled the 'next' button when the required fields are missing on the container page. I'm not sure what else should happen if the next step in the wizard isn't possible.

I'm going to work on fixing the issues with workload list views in a separate PR because I'd like to get this merged to stop blocking Wes from also addressing wizard-related concerns.

File: store/index.js [hunk 1]
```diff
@@ -291,7 +291,7 @@ export const actions = {
     });
 
     if ( getters['management/schemaFor'](COUNT) ) {
-      await dispatch('management/findAll', { type: COUNT, opt: { url: 'counts' } });
+      // await dispatch('management/findAll', { type: COUNT, opt: { url: 'counts' } });
```
Comment (vincent99) [inline store/index.js:294]: ?

Comment (mantis-toboggan-md) [reply store/index.js]: whoops, fixed

**Pull request merged**